src/record.rs

```diff
@@ -245,4 +226,47 @@ impl Header {
     fn crc32(&self) -> bincode::Result<u32> {
         self.to_raw().map(|raw| CRC32C.checksum(&raw))
     }
+
+    pub(crate) fn with_reversed_key_bytes(mut self) -> bincode::Result<Self> {
```

Comment:
This is not a common method, it is needed for migrations only, thus I think it is better to place it in `migration.rs`

Reply:
But I will need to make public `key` field and `update_checksum` method which is not good IMO

Reply:
Ok, I agree

Reply:
Done